docs/error-codes.md

```diff
@@ -75,63 +75,269 @@ the error code. For example:
 
-#### `IL2003`: Could not resolve 'assembly' assembly dependency specified in a 'PreserveDependency' attribute that targets method 'method'
+  ```XML
+  <linker>
+    <assembly fullname="test">
+      <type fullname="TestType" preserve="methods" />
+    </assembly>
+  </linker>
+  ```
+  ```C#
+  // IL2001: Type 'TestType' has no methods to preserve
+  struct TestType
+  {
+      public int Number;
+  }
+  ```
+
+#### `IL2003`: Could not resolve dependency assembly 'assembly name' specified in a 'PreserveDependency' attribute
 
 - The assembly 'assembly' in `PreserveDependency` attribute could not be resolved.
 
-#### `IL2004`: Could not resolve 'type' type dependency specified in a 'PreserveDependency' attribute that targets method 'method'
+  ```C#
+  // IL2003: Could not resolve dependency assembly 'NonExistentAssembly' specified in a 'PreserveDependency' attribute
+  [PreserveDependency("MyMethod", "MyType", "NonExistentAssembly")]
+  void TestMethod()
+  {
+  }
+  ```
 
-- The type 'type' in `PreserveDependency` attribute could not be resolved.
+#### `IL2004`: Could not resolve dependency type 'type' specified in a 'PreserveDependency' attribute
 
-#### `IL2005`: Could not resolve dependency member 'member' declared in type 'type' specified in a 'PreserveDependency' attribute that targets method 'method'
+- The type 'type' in `PreserveDependency` attribute could not be resolved.
 
-- The member 'member' in `PreserveDependency` attribute could not be resolved.
+  ```C#
+  // IL2004: Could not resolve dependency type 'NonExistentType' specified in a 'PreserveDependency' attribute
+  [PreserveDependency("MyMethod", "NonExistentType", "MyAssembly")]
+  void TestMethod()
+  {
+  }
+  ```
 
-#### `IL2006` (5): Unrecognized reflection pattern
+#### `IL2005`: Could not resolve dependency member 'member' declared in type 'type' specified in a 'PreserveDependency' attribute
 
-- The linker found an unrecognized reflection access pattern. The most likely reason for this is that the linker could not resolve a member that is being accessed dynamicallly. To fix this, use the `DynamicallyAccessedMemberAttribute` and specify the member kinds you're trying to access.
+- The member 'member' in `PreserveDependency` attribute could not be resolved.
 
-#### `IL2007`: Could not resolve assembly 'assembly' specified in the 'XML document location'
+  ```C#
+  // IL2005: Could not resolve dependency member 'NonExistentMethod' declared on type 'MyType' specified in a 'PreserveDependency' attribute
+  [PreserveDependency("NonExistentMethod", "MyType", "MyAssembly")]
+  void TestMethod()
+  {
+  }
+  ```
+
+#### `IL2006` Trim analysis: The requirements declared via the `DynamicallyAccessedMembersAttribute` on <value description> don't match those on <target description>. The source value must declare at least the same requirements as those declared on the target location it's assigned to
```

Comment:
What is the benefit of `Trim analysis` in the warning message for the developers?

Reply:
This is the subcategory - I'm adding it to this doc both as documentation and as a way to quickly review the subcategories. It will directly map to the `TrimAnalysis` property in MSBuild (which will suppress these messages).

Reply:
hmm, not sure this is best option. Maybe we could add Category bellow to not break the search

Reply:
I don't understand the concern? Is it about the ability to search this document?

Note, the output of the linker looks like this:
```
File.cs(11,22): Trim analysis warning IL2006: The requirements ....
```

Reply:
Yep, I think if I copy `IL2006: The requirements` text it should find me the doc about the warning. I don't see a value in having `Trim analysis` in the warning output. It just makes the output verbose and harder to read as different warnings have different formats and the information about Trim analysis is or could be already included in the warning code